apps/geo/views.py

```diff
@@ -241,5 +242,43 @@ def get(self, request, version=None):
             if not region.geo_options:
                 region.calc_cache()
             result[str(region.id)] = region.geo_options
-
         return response.Response(result)
+
+
+class GeoAreaView(viewsets.ReadOnlyModelViewSet):
+    permissions_classes = [permissions.IsAuthenticated]
+    serializer_class = GeoAreaSerializer
+
+    def get_queryset(self):
+        queryset = GeoArea.objects.prefetch_related(
+            'admin_level',
+            'admin_level__region'
+        ).distinct().annotate(
+            label=models.functions.Concat(
+                models.F('admin_level__title'),
+                models.Value('/'),
+                models.F('title'),
+                output_field=models.fields.CharField()
+            ),
+            region=models.F('admin_level__region_id'),
+            region_title=models.F('admin_level__region__title'),
+            admin_level_level=models.F('admin_level__level'),
+            admin_level_title=models.F('admin_level__title'),
+            key=models.F('id')
+        )
+        project = self.request.query_params.get('project')
+        if project:
+            project = Project.objects.get(id=project)
+            if not project.is_member(self.request.user):
+                raise exceptions.PermissionDenied()
+            return queryset.filter(
+                admin_level__region__project=project
```

Comment:
For filter let's use search filter backend or django_filter. It will be easier if we need to add additional fields and also for documentation.